pkg/cmd/adm/register_member.go

```diff
@@ -29,93 +38,177 @@ const (
 			term := ioutils.NewTerminal(cmd.InOrStdin, cmd.OutOrStdout)
-			ctx := clicontext.NewCommandContext(term, client.DefaultNewClient)
+			ctx := newExtendedCommandContext(term, client.DefaultNewClientFromRestConfig)
 			newCommand := func(name string, args ...string) *exec.Cmd {
 				return exec.Command(name, args...)
 			}
-			return registerMemberCluster(ctx, newCommand, hostKubeconfig, memberKubeconfig)
+			return registerMemberCluster(ctx, newCommand, hostKubeconfig, memberKubeconfig, useLetsEncrypt, 5*time.Minute)
 		},
 	}
 	defaultKubeconfigPath := ""
 	if home := homedir.HomeDir(); home != "" {
 		defaultKubeconfigPath = filepath.Join(home, ".kube", "config")
 	}
 	cmd.Flags().StringVar(&hostKubeconfig, "host-kubeconfig", defaultKubeconfigPath, "Path to the kubeconfig file of the host cluster (default: "+defaultKubeconfigPath+")")
-	flags.MustMarkRequired(cmd, "host-kubeconfig")
 	cmd.Flags().StringVar(&memberKubeconfig, "member-kubeconfig", defaultKubeconfigPath, "Path to the kubeconfig file of the member cluster (default: "+defaultKubeconfigPath+")")
-	flags.MustMarkRequired(cmd, "member-kubeconfig")
+	cmd.Flags().BoolVar(&useLetsEncrypt, "lets-encrypt", true, "Whether to use Let's Encrypt certificates or rely on the cluster certs (default: true)")
 	return cmd
 }
 
-func registerMemberCluster(ctx *clicontext.CommandContext, newCommand client.CommandCreator, hostKubeconfig, memberKubeconfig string) error {
-	ctx.AskForConfirmation(ioutils.WithMessagef("register member cluster from kubeconfig %s. Be aware that the ksctl disables automatic approval to prevent new users being provisioned to the new member cluster. "+
-		"You will need to enable it again manually.", memberKubeconfig))
+func registerMemberCluster(ctx *extendedCommandContext, newCommand client.CommandCreator, hostKubeconfig, memberKubeconfig string, useLetsEncrypt bool, waitForReadyTimeout time.Duration) error {
+	ctx.AskForConfirmation(ioutils.WithMessagef("register member cluster from kubeconfig %s. Note that the newly registered cluster will not be used for any space placement yet. This command will output an example"+
+		" SpaceProvisionerConfig that you can modify with the required configuration options and apply to make the cluster available for space placement.", memberKubeconfig))
 
-	hostClusterConfig, err := configuration.LoadClusterConfig(ctx, configuration.HostName)
+	// construct the client to the host cluster and figure out the API endpoint of the member cluster. We will use this information later,
+	// but let's obtain it first so that we fail without leaving garbage if there is some kind of simple misconfiguration.
+	hostConfig, err := clientcmd.LoadFromFile(hostKubeconfig)
 	if err != nil {
 		return err
 	}
-	hostClusterClient, err := ctx.NewClient(hostClusterConfig.Token, hostClusterConfig.ServerAPI)
+	hostClientConfig, err := clientcmd.NewDefaultClientConfig(*hostConfig, nil).ClientConfig()
 	if err != nil {
 		return err
 	}
```

Comment:
I'm using the `hostConfig`, i.e. the `cliendcmdapi.Config` representation of the kube config file to easily get at the API endpoint. The `rest.Config` obtained from `clientcmd.BuildConfigFromFlags` contains only the parsed version of the API endpoint that we would have to compose back. I thought it would be safer to just use the contents of the kube config file verbatim than try to reconstruct the information from it.